### PR TITLE
AP_GPS: NMEA add GSA msg support, allow nmea vdop

### PR DESCRIPTION
**Pull request by Jaaaky**
AP_GPS: NMEA add GSA msg support, allow nmea vdop

Ref http://freenmea.net/docs

I've tested this my self.

### CODE REVIEW

File: libraries/AP_GPS/AP_GPS_NMEA.cpp [hunk 1]
```diff
@@ -223,6 +223,13 @@ bool AP_GPS_NMEA::_have_new_message()
     if (_last_VTG_ms != 0) {
         _last_VTG_ms = 1;
     }
+    if (_last_GSA_ms != 0 &&
+        now - _last_GSA_ms > 150) {
```
Comment (tridge) [inline libraries/AP_GPS/AP_GPS_NMEA.cpp]: it looks like this makes GSA support compulsory? How can we be sure that all NMEA GPS modules support GSA?
I think we need to make this optional

**Comment by tridge** [comment]
we should also add GSA to our SITL implementation of NMEA so we can test this.

File: libraries/AP_GPS/AP_GPS_NMEA.cpp (hunk 1 above)
Comment (OXINARF) [reply libraries/AP_GPS/AP_GPS_NMEA.cpp]: It doesn't make it mandatory, this only enters if the message was received before (it's the same as VTG).

**Comment by Jaaaky** [comment]
@OXINARF 
For 2 reasons;
1. If GGA is missed
2. I've found that some receivers sends different info on messages where GSA is the most accurate. Specially num of used sats, for GGA I found on many the number doesn't go over 12, while counting it through GSA showed the real number.

**Comment by OXINARF** [comment]
@Jaaaky 

That doesn't answer my question (maybe you misread my question?).

For 1, if GGA is missing, you can use the same variables without any problem, no need for new ones.

For 2, I'm not sure we can apply that in a generic driver as NMEA is; even if we do, that still doesn't explain the new `*_alt` variables, you can just use the regular ones and then, on [GGA](https://github.com/ArduPilot/ardupilot/pull/11704/files#diff-0391c77a0ed972bb6a80621462b73883R262), check if `_last_GSA_ms` is different than 0 and don't set the variables that GSA already did - that's actually the only way to prevent GGA info to override the GSA, your code doesn't do that.

By the way, your code also prevents a GSA-only device from having HDOP and VDOP going up, it can only go down.

**Comment by peterbarker** [comment]
@Jaaaky are you still planning on pushing ahead with this?

**Pull request closed without merging**